src/SurveyDesign.jl

```diff
@@ -78,20 +78,25 @@ struct SurveyDesign <: AbstractSurveyDesign
         sampsize_labels = :sampsize
         data[!, sampsize_labels] = fill(length(unique(data[!, cluster])), (nrow(data),))
         if !(typeof(popsize) <: Nothing)
-            data[!, :weights] = data[!, popsize] ./ data[!, sampsize_labels]
-        elseif !(typeof(weights) <: Nothing)
-            data.weights = data[!, weights]
+            weights_labels = :weights
```

Comment:
I see that you are naming the column `weights` if the weights are not passed as a symbol and keep the given symbol otherwise. Are you trying to account for the fact that the user might provide a data frame that already has `weights` as a column name? In that case, this does not solve that. For example, the user might have:
```julia
julia> df = DataFrame(col1=[1, 2, 3], weights=[0.1, 0.3, 0.6])
3×2 DataFrame
 Row │ col1   weights
     │ Int64  Float64
─────┼────────────────
   1 │     1      0.1
   2 │     2      0.3
   3 │     3      0.6

julia> design = SurveyDesign(df; weights=:col1)
SurveyDesign:
data: 3×8 DataFrame
strata: none
cluster: none
popsize: [6, 6, 6  …  6]
sampsize: [3, 3, 3  …  3]
weights: [1, 2, 3  …  3]
probs: [1.0, 0.5, 0.333  …  0.333]

julia> design.data
3×8 DataFrame
 Row │ col1   weights  false_strata  false_cluster  sampsize  probs     allprobs  popsize
     │ Int64  Int64    String        Int64          Int64     Float64   Float64   Int64
─────┼────────────────────────────────────────────────────────────────────────────────────
   1 │     1        1  FALSE_STRATA              1         3  1.0       1.0             6
   2 │     2        2  FALSE_STRATA              2         3  0.5       0.5             6
   3 │     3        3  FALSE_STRATA              3         3  0.333333  0.333333        6
   ```
   
The original weights in the data were replaced because the user used another column for specifying the `weights` parameter.

I suggest renaming the column to `_weights`. It is highly unlikely that the user will provide a data frame with a column named `_weights`. What do you think?

Reply:
> Are you trying to account for the fact that the user might provide a data frame that already has `weights` as a column name? 

> I suggest renaming the column to `_weights`. It is highly unlikely that the user will provide a data frame with a column named `_weights`. What do you think?

While i think is a really edge case (1 out thousands of names for weight column), sure we can name it `_weights` or something else, since it doesnt matter to us because the accessor is always `design.weights`

Reply:
`_weights` is a good idea. Just to avoid that 0.1% chance of illusage of the package.

Reply:
we should append `_` to all the pointer symbols in the data structure. Similar logic for `popsize`, `sampsize` etc?

Reply:
done above